src/pages/Restaurants/Restaurant/Restaurant.tsx

```diff
@@ -0,0 +1,45 @@
+import { useState } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { useRestaurants } from '../../../utils/hooks/useRestaurants/useRestaurants';
+import { Meal, MealType, Restaurant as RestaurantProps } from '../../../utils/api/restaurantsService/restaurantsService';
+import RestaurantPopup from './RestaurantPopup/RestaurantPopup';
+import RestaurantImage from './RestaurantImage/RestaurantImage';
+import RestaurantDescription from './RestaurantDescription/RestaurantDescription';
+import MealsList from './MealsList/MealsList';
+import MealsFilter from './MealsFilter/MealsFilter';
+
+function Restaurant() {
+    const [isLiked, setIsLiked] = useState(false);
+    const [selectedMealTypes, setSelectedMealTypes] = useState<MealType[]>([]);
+    const navigate = useNavigate();
+    const params = useParams();
+    const { restaurantsFiltered } = useRestaurants();
+    const restaurant: RestaurantProps | undefined = restaurantsFiltered.find((restaurant) => restaurant.id === params.restaurantId);
```

Comment:
```suggestion
    const restaurant: RestaurantProps | null = restaurantsFiltered.find((restaurant) => restaurant.id === params.restaurantId);
```
Коллективное бессознательное пришло к таким установкам: когда видишь null, то понимаешь, что кто-то имел в виду, что будет null и сознательно его написал. А когда undefined - то подозреваешь, что кто-то что-то забыл или перепутал

Reply:
Попробовал null и тайпскрипт выдал ошибку, т.к. Array.find() возвращает undefined, если искомое не найдено 
![image](https://github.com/bronfood-com/front/assets/122983365/31965d34-098e-40db-870e-705a7f3dcb87)
Как быть?